010_ext-openssl-utf8-conversion-crash.patch
PHP_5_2
http://cvs.php.net/viewvc.cgi/php-src/ext/openssl/openssl.c?r1=1.98.2.5.2.51&r2=1.98.2.5.2.52&diff_format=u
http://bugs.php.net/bug.php?id=47828
Fix bug #47828 - Converting to UTF-8 can sometimes fail, check error codes and
avoid segfault.

diff -r e2602e8d2936 -r d409743914d7 ext/openssl/openssl.c
--- a/ext/openssl/openssl.c	Fri Apr 10 11:37:31 2009 +0200
+++ b/ext/openssl/openssl.c	Fri Apr 10 11:39:27 2009 +0200
@@ -301,7 +301,9 @@
 				str = X509_NAME_ENTRY_get_data(ne);
 				if (ASN1_STRING_type(str) != V_ASN1_UTF8STRING) {
 					to_add_len = ASN1_STRING_to_UTF8(&to_add, str);
-					add_next_index_stringl(subentries, (char *)to_add, to_add_len, 1);
+					if (to_add_len != -1) {
+						add_next_index_stringl(subentries, (char *)to_add, to_add_len, 1);
+					}
 				} else {
 					to_add = ASN1_STRING_data(str);
 					to_add_len = ASN1_STRING_length(str);
@@ -317,7 +319,7 @@
 		} else {
 			zval_dtor(subentries);
 			FREE_ZVAL(subentries);
-			if (obj_cnt && str) {
+			if (obj_cnt && str && to_add_len > -1) {
 				add_assoc_stringl(subitem, sname, (char *)to_add, to_add_len, 1);
 			}
 		}
